backport_packages/refactor_backport_packages.py

```diff
@@ -0,0 +1,786 @@
+             from airflow.providers.dingding.hooks.dingding import DingdingHook
+             from airflow.utils.decorators import apply_defaults
+
+        """
+        changes = [
+            ("airflow.operators.bash", "airflow.operators.bash_operator"),
+            ("airflow.operators.python", "airflow.operators.python_operator"),
+            ("airflow.utils.session", "airflow.utils.db"),
+            (
+                "airflow.providers.cncf.kubernetes.operators.kubernetes_pod",
+                "airflow.contrib.operators.kubernetes_pod_operator"
+            ),
+        ]
+        for new, old in changes:
+            self.qry.select_module(new).rename(old)
+
+    def add_provide_context_to_python_operators(self) -> None:
+        """
+
+        Adds provide context to usages of Python/BranchPython Operators in example dags.
+        Note that those changes  apply to example DAGs not to the operators/hooks erc.
+        We package the example DAGs together with the provider classes and they should serve as
+        examples independently on the version of Airflow it will be installed in.
+        Provide_context feature in Python operators was feature added 2.0.0 and we are still
+        using the "Core" operators from the Airflow version that the backport packages are installed
+        in - the "Core" operators do not have (for now) their own provider package.
+
+        The core operators are:
+
+            * Python
+            * BranchPython
+            * Bash
+            * Branch
+            * Dummy
+            * LatestOnly
+            * ShortCircuit
+            * PythonVirtualEnv
+
+
+        Example diff generated:
+
+        .. code-block:: diff
+
+            --- ./airflow/providers/amazon/aws/example_dags/example_google_api_to_s3_transfer_advanced.py
+            +++ ./airflow/providers/amazon/aws/example_dags/example_google_api_to_s3_transfer_advanced.py
+            @@ -105,7 +105,8 @@
+                         task_video_ids_to_s3.google_api_response_via_xcom,
+                         task_video_ids_to_s3.task_id
+                     ],
+            -        task_id='check_and_transform_video_ids'
+            +        task_id='check_and_transform_video_ids',
+            +        provide_context=True
+                 )
+
+        """
+        # noinspection PyUnusedLocal
+        def add_provide_context_to_python_operator(node: LN, capture: Capture, filename: Filename) -> None:
+            fn_args = capture['function_arguments'][0]
+            fn_args.append_child(Comma())
+
+            provide_context_arg = KeywordArg(Name('provide_context'), Name('True'))
+            provide_context_arg.prefix = fn_args.children[0].prefix
+            fn_args.append_child(provide_context_arg)
+
+        (
+            self.qry.
+            select_function("PythonOperator").
+            is_call().
+            is_filename(include=r"mlengine_operator_utils.py$").
```

Comment:
Can this be broadeded to select any example dag instead of just this one specific one?

Reply:
Actually. Yeah. I am not sure at all why it was limited in the first place (I have not noticed that). I removed the limitations and I will see whether we have any problems. I do not expect to. Good catch